extension/src/pipeline/util.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  getFileExtension,
+  loadJson,
+  loadCsv,
+  loadTsv,
+  loadYamlAsJs
+} from '../fileSystem'
+import { quickPickOne } from '../vscode/quickPick'
+import { pickFile } from '../vscode/resourcePicker'
+import { Title } from '../vscode/title'
+
+const parseDataFile = (file: string) => {
+  const ext = getFileExtension(file)
+
+  if (ext === '.json') {
+    return loadJson<Record<string, unknown> | unknown[]>(file)
+  }
+
+  if (ext === '.csv') {
+    return loadCsv(file)
+  }
+
+  if (ext === '.tsv') {
+    return loadTsv(file)
+  }
+
+  if (ext === '.yaml') {
+    return loadYamlAsJs<Record<string, unknown>>(file)
+  }
+}
+
+const pickDataFile = () => {
+  return pickFile(Title.SELECT_PLOT_DATA, {
+    filters: {
+      'Data Formats': ['json', 'csv', 'tsv', 'yaml']
+    },
+    openLabel: 'Select'
```

Comment:
[Q] Should this `openLabel` option be set to "Select" whenever we use `pickFile`?